doc/contribute/contributor_expectations.rst

```diff
@@ -0,0 +1,280 @@
+  Contributors may mark a PR as draft and explicitly request reviewers to
+  provide early feedback, even with failing CI checks.
+
+- When breaking a PR into multiple commits, each commit must build cleanly. The
+  CI system does not enforce this policy, so it is the PR author's
+  responsibility to verify.
+
+- When major new functionality is added, tests for the new functionality shall
+  be added to the automated test suite. All API functions should have test cases
+  and there should be tests for the behavior contracts of the API. Maintainers
+  and reviewers have the discretion to determine if the provided tests are
+  sufficient. The examples below demonstrate best practices on how to test APIs
+  effectively.
+
+    - :zephyr_file:`Kernel timer tests <tests/kernel/timer/timer_behavior>`
+      provide around 85% test coverage for the
+      :zephyr_file:`kernel timer <kernel/timer.c>`, measured by lines of code.
+    - Emulators for off-chip peripherals are an effective way to test driver
+      APIs. The :zephyr_file:`fuel gauge tests <tests/drivers/fuel_gauge/sbs_gauge>`
+      use the :zephyr_file:`smart battery emulator <drivers/sensor/sbs_gauge/emul_sbs_gauge.c>`,
+      providing test coverage for the
+      :zephyr_file:`fuel gauge API <include/zephyr/drivers/fuel_gauge.h>`
+      and the :zephyr_file:`smart battery driver <drivers/fuel_gauge/sbs_gauge/sbs_gauge.c>`.
+    - Code coverage reports for the Zephyr project are available on `Codecov`_.
+
+- Incompatible changes to APIs must also update the release notes for the
+  next release detailing the change.  APIs marked as experimental are excluded
+  from this requirement.
+
+- Changes to APIs must increment the API version number according to the API
+  version rules.
+
+- PRs must also satisfy all :ref:`merge_criteria` before a member of the release
+  engineering team merges the PR into the zephyr tree.
+
+Maintainers may request that contributors break up a PR into smaller PRs and may
+request that they create an :ref:`RFC proposal <rfcs>`.
+
+.. _`Codecov`: https://app.codecov.io/gh/zephyrproject-rtos/zephyr
+
+Workflow Suggestions That Help Reviewers
+========================================
+
+- Unless they applied the reviewer's recommendation exactly, authors must not
+  resolve and hide comments, they must let the initial reviewer do it. The
+  Zephyr project does not require all comments to be resolved before merge.
+  Leaving some completed discussions open can sometimes be useful to understand
+  the greater picture.
+
+- Respond to comments using the "Start Review" and "Add Review" green buttons in
+  the "Files changed" view. This allows responding to multiple comments and
+  publishing the responses in bulk. This reduces the number of emails sent to
+  reviewers.
+
+- As GitHub does not implement |git range-diff|_, try to minimize rebases in the
+  middle of a review. If a rebase is required, push this as a separate update
```

Comment:
> As GitHub does not implement |git range-diff|_, try to minimize rebases in the  middle of a review.

Could an admin please disable the "Update branch" button that shows up in every PR?

It's apparently just one checkbox:
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-suggestions-to-update-pull-request-branches

Reply:
> Could an admin please disable the "Update branch" button that shows up in every PR?

Done